### PR TITLE
Enable SQL Auth Provider support

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Addresses #13931 #19905 

_Companion PR: https://github.com/microsoft/sqltoolsservice/pull/1860_
Introduces new setting that works in-tune with SQLToolsService:

image
This setting is disabled by default.

Pending tasks _[for next PRs - non blocking]:_
- [ ] Prompt to reload ADS when setting is changed. [@cheenamalhotra]
- [ ] Skip calls to Azure Core extension to fetch access token when this feature is enabled from different paths. [@cheenamalhotra]

### CODE REVIEW

File: extensions/mssql/src/utils.ts
Comment (Charles-Gagnon) [inline extensions/mssql/src/utils.ts:143]: This shouldn't be true anymore - I had it default to dev so there should always be a value for this. 

https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/api/common/sqlExtHost.api.impl.ts#L623

Looks like I just forgot to update the comment above. Could you verify this is true and then remove these comments if so?

File: extensions/mssql/package.nls.json
Comment (Charles-Gagnon) [inline extensions/mssql/package.nls.json:178]: I think this reads a little better. 

Is there some docs you can link to for this? I'm not sure most people are going to understand what this setting means so providing some way for them to find out more if they're curious would be nice. 

```suggestion
	"mssql.enableSqlAuthenticationProvider" : "Enables the Sql Authentication Provider for 'Active Directory Interactive' authentication mode when user selects 'AzureMFA' authentication. This enables Server-side resource endpoint integration when fetching access tokens.",
```

File: extensions/mssql/src/utils.ts
Comment (Charles-Gagnon) [inline extensions/mssql/src/utils.ts:136]: This doesn't seem to need to be async (if you could fix up the other one too that'd be great, another thing I missed cleaning up when I changed this)

**Review by kburtram — APPROVED** [review]

**Comment by kburtram** [comment]
You should probably wait until the STS PR is merged and then include the STS vbump with or before this change.

File: extensions/mssql/package.nls.json
Comment (cheenamalhotra) [reply extensions/mssql/package.nls.json:178]: Not sure if this is something that should be explained further.. this is between driver and tools and how we implemented it, and this design is expected from our tools IMO. End users shouldn't have to investigate why they couldn't connect to certain endpoints as this was our design gap.

This configuration is only until we get confidence and stability in design and enable it by default - it will be removed in future.

File: extensions/mssql/src/sqlToolsServer.ts
Comment (Charles-Gagnon) [inline extensions/mssql/src/sqlToolsServer.ts:126]: Actually - is this something that can be toggled during runtime? Because STS already gets notified when settings are changed (https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/Workspace/WorkspaceService.cs#L326) so you could just hook into that instead and then remove the need for the app restart. That doesn't get the app quality - but that'd be easy enough to pass up as a parameter instead of this.

Comment (cheenamalhotra) [reply extensions/mssql/src/sqlToolsServer.ts:126]: I don't think it's a safe toggle, since it'll also change the way Access tokens are sent to STS and refreshed, it will make existing connections unstable. I would rather force a reload when this is enabled.

File: extensions/azurecore/src/account-provider/auths/azureAuth.ts
Comment (Charles-Gagnon) [inline extensions/azurecore/src/account-provider/auths/azureAuth.ts:207]: Make sure the expected values of azureResource are documented so callers know what the string is expected to be (same for other functions that you've changed this for)

File: extensions/mssql/src/utils.ts
Comment (Charles-Gagnon) [inline extensions/mssql/src/utils.ts:173]: This should be a runtime toggle, we shouldn't be continuing to collect PII if the user has this turned off but hasn't reloaded the app yet

**Pull request merged**
